plagiarism/http.go

```diff
@@ -0,0 +1,112 @@
+package plagiarism
+
+import (
+    "net/http"
+    "context"
+    "time"
+    "encoding/json"
+    "os/signal"
+    "os"
+    "log"
+)
+
+// Response struct will be used to transform to JSON format
+type Response struct {
+    Status string
+    Results map[string]string
+}
+
+// Server init variable
+type Server struct {
+    SubmissionDir string
+    ResultDir string
+    ID uint32
+    Logger *log.Logger
+    Port string
+}
+
+
+// Handler GET and POST in server
+func (s *Server) Handler() http.Handler {
+    return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+        // gluons only supports GET and POST reques to / path
+        // Throw HTTP error when path other than / is requested
+        if r.URL.Path != "/" {
+            http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+            return
+        }
+
+        // Decode http request into variable
+        guids, err := DecodeRequest(r.Body)
+        if err != nil {
+            http.Error(w, http.StatusText(http.StatusNotAcceptable), http.StatusNotAcceptable)
+        }
+
+        // Initialize analysis struct
+        analysis := Analysis{
+            Guids: guids,
+            SubmissionDir: s.SubmissionDir,
+            Mossid: s.ID,
+            Logger: s.Logger,
+            ResultDir: s.ResultDir,
+        }
+        var response Response
+        switch r.Method {
+        case "GET":
+            response = analysis.GetResults()
+        case "POST":
+            response = analysis.RunDetector()
+        default:
+            http.Error(w, http.StatusText(http.StatusNotAcceptable), http.StatusNotAcceptable)
+        }
+        w.Header().Set("Content-Type", "application/json; charset=utf-8")
+        w.WriteHeader(http.StatusOK)
+        json.NewEncoder(w).Encode(response)
+    })
+}
+
+
+// InitServer start golang server
+func (s *Server) InitServer() {
+    router := http.NewServeMux()
+    router.Handle("/", s.Handler())
+    server := &http.Server{
+        Addr:           ":"+s.Port,
+        Handler:        s.Handler(),
+        ErrorLog:       s.Logger,
+        ReadTimeout:    5 *  time.Second,
+        WriteTimeout:   10 * time.Second,
+        IdleTimeout:    15 * time.Second,
+    }
+
+    done := make(chan bool)
+    quit := make(chan os.Signal, 1)
+
+    // Stop when sigint is triggered
+    signal.Notify(quit, os.Interrupt)
```

Comment:
It's better if you move all the top-level signal handling and stuff to the main file. Similar to https://github.com/omegaup/gitserver/blob/76ddfcd9f566a7d0ca4726f43aec4ec69b9c817b/cmd/omegaup-gitserver/main.go#L53

So most of this function can actually be used as-is in `main()`. Try to follow the patterns used in that link as closely as possible, for consistency.